src/lib.rs

```diff
@@ -278,6 +299,39 @@ impl hal::blocking::spi::Write<u8> for Spidev {
     }
 }
 
+pub use hal::blocking::spi::{Operation as SpiOperation};
+
+impl hal::blocking::spi::Transactional<u8> for Spidev {
+    type Error = io::Error;
+
+    fn exec<'a, O>(&mut self, mut operations: O) -> Result<(), Self::Error>
+        where O: AsMut<[SpiOperation<'a, u8>]> {
+
+        // Map types from generic to linux objects
+        let mut messages: Vec<_> = operations.as_mut().iter_mut().map(|a| {
+            match a {
+                SpiOperation::Write(w) => SpidevTransfer::write(w),
+                SpiOperation::WriteRead(r) => {
+                    // TODO: is spidev okay with the same array pointer
+                    // being used twice? If not, need some kind of vector 
+                    // pool that will outlive the transfer
+                    let w = unsafe {
```

Comment:
@posborne is this too cursed? i can't seem to find any documentation as to whether it's okay or not

Reply:
From looking at the [implementation](https://github.com/torvalds/linux/blob/aae1464f46a2403565f75717438118691d31ccf1/drivers/spi/spidev.c#L198) I think it would work, as the buffers are copied in and out. However, a documented guarantee would be better.

Reply:
As @eldruin notes, this should be fine with how the kernel works (performing a copy_from_user).  That being said, I do agree that it does feel off, although I can't reason about any way it could go wrong in practice.

I would be OK with landing this here and considering adding an API to spidev which would be designed to explicitly provide safe semantics for this way of interacting with the syscall for future use.  That is, something like (not sure about naming):

```
pub fn read_write_shared(txrx_buf: &'a mut [u8]) -> Self
```